tests/pthread/main_thread_join.cpp

```diff
@@ -32,8 +33,12 @@ void *ThreadMain(void *arg) {
   // Delay to force the main thread to try and fail a few times before
   // succeeding.
   while (tries.load() < EXPECTED_TRIES) {}
+#else
+  // Delay a bit to force the main thread to actually block, as exited
+  // threads can be joined without blocking.
+  emscripten_thread_sleep(1000);
```

Comment:
Is this change actually needed?  What happens if you don't make this change?  Does the test fail?

Reply:
As far as I could remember, the test did indeed fail, because exited threads can be joined without any blocking now. Let me re-check that.

Reply:
I believe we should be calling `emscripten_check_blocking_allowed` even when a thread can be joined without blocking.  The idea is to warn folks, even in the cases when they got lucky with the timing.

Reply:
i.e. the warning/error should still show even when joining already exited threads.

Reply:
Reverted that with commit 44776e80a95f40770da7b4ee5f6d55f75c84f092. PR description has been updated accordingly.